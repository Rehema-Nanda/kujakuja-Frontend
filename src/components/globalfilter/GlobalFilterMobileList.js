import React, { useState } from "react";
import PropTypes from "prop-types";
import GlobalFilterDropdownSearch from "./GlobalFilterDropdownSearch";
import GlobalFilterCheckbox from "./GlobalFilterCheckbox";

const GlobalFilterMobileList = (props) => {
    const {
        items,
        itemLabelProperty,
        selectItemHandler,
        selectItemHandlerArgs,
        selectedItems,
        showSearch,
        allowMultipleServicePointSelect,
    } = props;

    const [itemsToDisplay, setItemsToDisplay] = useState(items);

    return (
        <div>
            {showSearch
            && (
                <GlobalFilterDropdownSearch
                    className="mobile-list-search"
                    items={items}
                    itemLabelProperty={itemLabelProperty}
                    setResults={setItemsToDisplay}
                />
            )}
            <ul className="map-controls-popover-list fadein">
                {itemsToDisplay.map((item) => {
                    return (
                        <li key={item.id}>
                            <GlobalFilterCheckbox
                                item={item}
                                itemLabelProperty={itemLabelProperty}
                                selectItemHandler={selectItemHandler}
                                selectItemHandlerArgs={selectItemHandlerArgs}
                                selectedItems={selectedItems}
                                allowMultipleServicePointSelect={allowMultipleServicePointSelect}
                            />
                        </li>
                    );
                })}
            </ul>
        </div>
    );
};

GlobalFilterMobileList.propTypes = {
    items: PropTypes.arrayOf(PropTypes.object).isRequired,
    itemLabelProperty: PropTypes.string.isRequired,
    selectItemHandler: PropTypes.func.isRequired,
    selectItemHandlerArgs: PropTypes.arrayOf(PropTypes.any),
    selectedItems: PropTypes.arrayOf(PropTypes.object).isRequired,
    showSearch: PropTypes.bool,
    allowMultipleServicePointSelect: PropTypes.bool,
};

GlobalFilterMobileList.defaultProps = {
    selectItemHandlerArgs: [],
    showSearch: false,
    allowMultipleServicePointSelect: true,
};

export default GlobalFilterMobileList;
