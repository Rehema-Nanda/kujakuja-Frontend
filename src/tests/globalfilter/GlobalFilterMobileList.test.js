import React from "react";
import renderer from "react-test-renderer";
import GlobalFilterMobileList from "../../components/globalfilter/GlobalFilterMobileList";

it("renders correctly", () => {
    const props = {
        items: [],
        itemLabelProperty: "name",
        selectItemHandler: jest.fn(),
        selectedItems: [],
    };

    const tree = renderer.create(
        <GlobalFilterMobileList {...props} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
