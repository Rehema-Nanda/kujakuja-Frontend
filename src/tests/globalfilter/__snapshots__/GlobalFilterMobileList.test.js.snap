// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <ul
    className="map-controls-popover-list fadein"
  />
</div>
`;
